cedar-dafny/validation/types.dfy

```diff
@@ -20,6 +20,16 @@ module validation.types {
   import opened def.base
   import opened def.core
 
+  // The ValidationMode determines whether to use permissive or strict typechecking
+  datatype ValidationMode = Permissive | Strict {
+    predicate isStrict() {
```

Comment:
I wonder if it makes sense to use the `Strict?` method Dafny automatically generates instead.

Reply:
I think you're right